runtime/libpgmath/lib/CMakeLists.txt

```diff
@@ -116,7 +116,7 @@ if(${LIBPGMATH_SYSTEM_PROCESSOR} MATCHES "x86_64" AND NOT ${LIBPGMATH_WITH_GENER
   else()
     message(FATAL "Operating System not supported.")
   endif()
-elseif(${LIBPGMATH_SYSTEM_PROCESSOR} MATCHES "ppc64le")
+elseif(${LIBPGMATH_SYSTEM_PROCESSOR} MATCHES "ppc64le" AND NOT ${LIBPGMATH_WITH_GENERIC})
```

Comment:
Instead of adding `AND NOT ${LIBPGMATH_WITH_GENERIC}` to every test of the multi-branch `if()` block, consider testing `if (${LIBPGMATH_WITH_GENERIC})` first, and simplifying the conditions in the subsequent tests.

Reply:
This pattern is repeated in a couple of other files below.

Reply:
Sadly, the way PPC64 architecture is handled makes it harder to do it ANY better (starting from the fact PPC64 logic introduces additional variable, `PROCESSOR` next to `LIBPGMATH_SYSTEM_PROCESSOR`). I do not have any ppc64 Linux target that I could use to verify correctness of any of my changes to this logic. Therefore, there are two options:
- considering my current patch as it is now
- calling volunteers with PPC64 targets for help